sdk/core/Azure.Core/src/Internal/DateTimeOffsetHelpers.netfx.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Azure
+{
+    // https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.DateTime.netfx.cs
+    internal static partial class DateTimeOffsetHelpers
+    {
+        private static TimeSync _timeSync = new TimeSync();
+
+        // sync DateTime and Stopwatch ticks every 2 hours
+#pragma warning disable CA1823 // suppress unused field warning, as it's used to keep the timer alive
+        private static readonly Timer _syncTimeUpdater = InitializeSyncTimer();
+#pragma warning restore CA1823
+
+        private static TimerCallback _sync = s => 
+        {
+            // wait for DateTime.UtcNow update to the next granular value
+            Thread.Sleep(1);
+            Interlocked.Exchange(ref _timeSync, new TimeSync());
+        };
+
+        /// <summary>
+        /// Returns high resolution (1 DateTime tick) current UTC DateTime.
+        /// </summary>
+        private static DateTimeOffset GetUtcNowImplementation()
+        {
+            // DateTime.UtcNow accuracy on .NET Framework is ~16ms, this method
```

Comment:
Why do we need more than 16ms accuracy ?

Reply:
This has been causing one of the tests to fail from time to time. Also, I'll need this to be able to correctly set start time of the activity in another changeset.

Reply:
Can we fix the test instead? Seems like a lot of extra production code to pull in to fix the test.

Reply:
Fixed